components/datadog/agent/ecsFargate.go

```diff
@@ -9,15 +9,11 @@ import (
 	"github.com/DataDog/test-infra-definitions/components/datadog/fakeintake"
 )
 
-func ECSFargateLinuxContainerDefinition(e config.Env, image string, apiKeySSMParamName pulumi.StringInput, fakeintake *fakeintake.Fakeintake, logConfig ecs.TaskDefinitionLogConfigurationPtrInput) *ecs.TaskDefinitionContainerDefinitionArgs {
-	if image == "" {
-		image = dockerAgentFullImagePath(e, "public.ecr.aws/datadog/agent", "latest", false)
-	}
-
+func ECSFargateLinuxContainerDefinition(e config.Env, apiKeySSMParamName pulumi.StringInput, fakeintake *fakeintake.Fakeintake, logConfig ecs.TaskDefinitionLogConfigurationPtrInput) *ecs.TaskDefinitionContainerDefinitionArgs {
 	return &ecs.TaskDefinitionContainerDefinitionArgs{
 		Cpu:       pulumi.IntPtr(0),
 		Name:      pulumi.String("datadog-agent"),
-		Image:     pulumi.String(image),
+		Image:     pulumi.String(dockerAgentFullImagePath(e, "public.ecr.aws/datadog/agent", "latest", false)),
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
Given `dockerAgentFullImagePath` [implementation](https://github.com/DataDog/test-infra-definitions/blob/pducolin/reset-ssh-auth-keys-windows/components/datadog/agent/docker_image.go#L20-L57)
```go
func dockerAgentFullImagePath(e config.Env, repositoryPath, imageTag string, otel bool) string {
	if e.AgentFullImagePath() != "" {
		return e.AgentFullImagePath()
	}

	// if agent pipeline id and commit sha are defined, use the image from the pipeline pushed on agent QA registry
	if e.PipelineID() != "" && e.CommitSHA() != "" && imageTag == "" {
		tag := fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA())
		if otel {
			tag = fmt.Sprintf("%s-7-ot-beta", tag)
		}

		exists, err := e.InternalRegistryImageTagExists(fmt.Sprintf("%s/agent", e.InternalRegistry()), tag)
		if err != nil || !exists {
			panic(fmt.Sprintf("image %s/agent:%s not found in the internal registry", e.InternalRegistry(), tag))
		}
		return utils.BuildDockerImagePath(fmt.Sprintf("%s/agent", e.InternalRegistry()), tag)
	}

	...
```
and given we pass [pipeline id and commit sha](https://github.com/DataDog/datadog-agent/blob/auto-bump/bump-test-infra-ef9171b0/.gitlab/e2e/e2e.yml#L40-L41) from the CI config in `datadog-agent`, I expect this to still install the latest agent version from "public.ecr.aws/datadog/agent". 

In order to install the pipeline version and fallback to latest by default, I expect this to be
```suggestion
		Image:     pulumi.String(dockerAgentFullImagePath(e, "", "", false)),
```

Reply:
I see how `dockerAgentFullImagePath` can be confusing for the caller. Probably worth renaming it and add more context in the function description, will track this on our side so that we can follow up